relentless/simulate/hoomd.py

```diff
@@ -801,6 +801,53 @@ def reset(self):
             # rdf will be reset on next call
 
 
+class WriteTrajectory(simulate.AnalysisOperation):
+    """Writes a .gsd file.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the .gsd file to be written.
+    every : int
+        Interval of time steps at which to write a snapshot of the simulation.
+    velocity : bool
+        Log particle velocity.
+    image : bool
+        Log particle image.
+    typeid : bool
+        Log particle type ID.
+    mass : bool
+        Log particle mass.
+
+    """
+
+    def __init__(self, filename, every, velocity, image, typeid, mass):
+        self.filename = filename
+        self.every = every
+        self.velocity = velocity
+        self.image = image
+        self.typeid = typeid
+        self.mass = mass
+
+    def __call__(self, sim):
+        _dynamic = ["property"]
+        if self.velocity is True or self.image is True:
+            _dynamic.append("momentum")
+
+        with sim.hoomd:
+            all_ = hoomd.group.all()
+            hoomd.dump.gsd(
+                self.filename,
```

Comment:
This needs to be wrapped by the directory most of the time:
```suggestion
                sim.directory.file(self.filename),
```
We should decide if we want to optionally allow an absolute path that does not do this.

We should also make sure `sim.directory` is not None.